crates/primitives/src/l1.rs

```diff
@@ -227,10 +227,13 @@ pub struct L1Status {
     pub cur_tip_blkid: String,
 
     /// Last published txid where L2 blob was present
-    pub last_published_txid: Option<String>,
+    pub last_published_txid: Option<Txid>,
```

Comment:
Yes! Every time that we use `String` or `Vec<u8>` instead of proper types a puppy dies.